src/services/EdFi.DataManagementService.Api/Core/Validation/DocumentValidator.cs

```diff
@@ -103,16 +103,53 @@ PruneResult PruneOverpostedData(JsonNode? documentBody, EvaluationResults evalua
                 }
                 if (detail.Errors != null && detail.Errors.Any())
                 {
-                    foreach (var error in detail.Errors)
+                    foreach (var error in detail.Errors.Select(x => x.Value))
                     {
-                        val.Add($"{propertyName}{error.Value}");
-                        validationErrors.Add(propertyName, val.ToArray());
+                        var splitErrors = SplitErrorDetail(error, propertyName);
+
+                        foreach (var splitError in splitErrors)
+                        {
+                            if (validationErrors.ContainsKey(splitError.Key))
+                            {
+                                var existingErrors = validationErrors[splitError.Key].ToList();
+                                existingErrors.AddRange(splitError.Value);
+                                validationErrors[splitError.Key] = existingErrors.ToArray();
+                            }
+                            else
+                            {
+                                validationErrors.Add(splitError.Key, splitError.Value);
+                            }
+                        }
                     }
                 }
             }
             return validationErrors;
         }
     }
+
+    private static Dictionary<string, string[]> SplitErrorDetail(string error, string propertyName)
+    {
+        var validations = new Dictionary<string, string[]>();
+        if (error.Contains("[") && error.Contains("]"))
+        {
+            MatchCollection hits = Regex.Matches(error, "\"([^\"]*)\"");
```

Comment:
Suggestion: move this regex to a static property and set the compiled flag. That will make this have higher performance. I think there is another example of that somewhere in the application - just look for other examples of `Regex` in the solution.